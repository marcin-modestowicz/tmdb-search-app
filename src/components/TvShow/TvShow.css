.root {
  box-sizing: border-box;
  margin-top: 1em;
  padding: 1em 1em;
  width: 100%;
  max-width: 40em;
  font-family: sans-serif;
}

.header h3,
.header h4,
.header h5,
.header h6 {
  margin-top: 0;
}

.content > img {
  float: left;
  margin: 0 1em 1em 0;
  width: 50%;
  max-width: 185px;
}

.content p {
  margin: 0;
  line-height: 1.5em;
}
