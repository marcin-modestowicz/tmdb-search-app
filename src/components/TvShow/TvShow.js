// @flow

import React, { Component } from "react";
import Image from "../Image";
import styles from "./TvShow.css";

type Props = {
  tvShow: Object
};

const TvShow = ({ tvShow }: Props) => {
  const {
    vote_average,
    name,
    poster_path,
    original_name,
    overview,
    first_air_date
  } = tvShow;
  const hasOriginalName = name !== original_name;
  const hasPoster = poster_path != null;

  return (
    <div className={styles.root}>
      <header className={styles.header}>
        <h3>{name}</h3>
        {hasOriginalName && <h4>{original_name}</h4>}
        <h5>
          First air date: <time>{first_air_date}</time>
        </h5>
        <h6>Rating: {vote_average}</h6>
      </header>
      <article className={styles.content}>
        {hasPoster && <Image name={poster_path} />}
        <p>{overview}</p>
      </article>
    </div>
  );
};

export default TvShow;
