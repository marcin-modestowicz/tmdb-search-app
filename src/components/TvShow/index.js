// @flow
export { default } from "./TvShow";
