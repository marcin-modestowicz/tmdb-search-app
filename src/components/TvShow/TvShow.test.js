// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TvShow from "./TvShow";

describe("TvShow component", () => {
  test("should render properly", () => {
    const tvShowData = {
      original_name: "Saturday Night Live",
      name: "Saturday Night Live",
      vote_average: 6.43,
      poster_path: "/aXdHYUXubjdVzdQvUsV5OxpCNeu.jpg",
      first_air_date: "1975-10-11",
      overview:
        "A late-night live television sketch comedy and variety show created by Lorne Michaels. The show's comedy sketches, which parody contemporary culture and politics, are performed by a large and varying cast of repertory and newer cast members. Each episode is hosted by a celebrity guest, who usually delivers an opening monologue and performs in sketches with the cast, and features performances by a musical guest."
    };
    const tvShow = shallow(<TvShow tvShow={tvShowData} />);
    expect(toJson(tvShow)).toMatchSnapshot();
  });
});
