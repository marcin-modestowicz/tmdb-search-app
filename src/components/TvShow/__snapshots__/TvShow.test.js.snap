// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShow component should render properly 1`] = `
<div
  className="root"
>
  <header
    className="header"
  >
    <h3>
      Saturday Night Live
    </h3>
    <h5>
      First air date: 
      <time>
        1975-10-11
      </time>
    </h5>
    <h6>
      Rating: 
      6.43
    </h6>
  </header>
  <article
    className="content"
  >
    <Image
      name="/aXdHYUXubjdVzdQvUsV5OxpCNeu.jpg"
    />
    <p>
      A late-night live television sketch comedy and variety show created by Lorne Michaels. The show's comedy sketches, which parody contemporary culture and politics, are performed by a large and varying cast of repertory and newer cast members. Each episode is hosted by a celebrity guest, who usually delivers an opening monologue and performs in sketches with the cast, and features performances by a musical guest.
    </p>
  </article>
</div>
`;
